### PR TITLE
[#13099][feat] Support Wan 2.2 5B TI2V model

### PR DESCRIPTION
**Pull request by abc99lr**
## Release Notes

* **New Features**
  * Added support for Wan 2.2 TI2V-5B image-to-video generation
  * Image input support for visual generation pipeline
  * Enhanced Wan 2.2 model variants (14B and 5B sizes)

* **Improvements**
  * Improved latent processing and timestep handling for image-to-video workflows
  * Refined configuration for different Wan 2.2 model variants

## Description
The Wan 2.2 5B model is a unified text-image-to-video model that handles both T2V and I2V in a single pipeline. Supports #13099 

- Detect the Wan 2.2 5B model via `expand_timesteps` flag in `model_index.json`
- Build Wan 2.2 5B's per-patch 2D timestep during denoising: frame-0 patches get t=0 (I2V) or all patches get current_t (T2V)
- Extend `WanTransformer3DModel` to accept per-patch `timestep` tensors `([B, num_patches])` alongside the standard scalar/batch form
- Add `post_step_fn` callback to `denoise()` to pin frame 0 back to the VAE-encoded input image after each scheduler step. This is needed for Wan 2.2 5B's I2V
- Add `_WAN22_5B_PARAMS` with `704×1280` native resolution, `40` steps, guidance `5.0`
- Rename `is_wan22` to `is_wan22_14b` throughout for clarity; remove unnecessary  `num_inference_steps` & `guidance_scale` fallbacks from forward()

@o-stoner @chang-l

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_utils.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Optional
+
+import torch
+
+
+def retrieve_latents(
```
Comment (o-stoner) [inline tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_utils.py:20]: @chang-l we don't have a utils file for other pipelines currently; do you agree we should start this per-pipeline utils pattern in VisualGen or move these utils elsewhere?

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py [hunk 2]
```diff
@@ -79,13 +82,16 @@
...
     def __init__(self, model_config):
-        # Wan2.2 two-stage denoising parameters
+        # Wan2.2 A14B two-stage denoising parameters
         self.transformer_2 = None
         self.boundary_ratio = getattr(model_config.pretrained_config, "boundary_ratio", None)
-        self.is_wan22 = self.boundary_ratio is not None
+        self.expand_timesteps = getattr(model_config.pretrained_config, "expand_timesteps", False)
+        # Derived model type flags
+        self.is_wan22_14b = self.boundary_ratio is not None
+        self.is_wan22_5b = self.expand_timesteps
```
Comment (o-stoner) [inline tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py:91]: Do we want to create any pipeline sanity test files for Wan TI2V like the Wan T2V/I2V ones [here](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests/unittest/_torch/visual_gen)?

**Review by o-stoner — APPROVED** [review]
Overall LGTM! Note, we will want to add Wan 2.2 TI2V 5B as a supported model in our documentation [here](https://github.com/NVIDIA/TensorRT-LLM/blob/main/docs/source/models/visual-generation.md). Thanks for the PR

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_utils.py (hunk 1 above)
Comment (abc99lr) [reply tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_utils.py:20]: Hi @chang-l , could you please comment on this question?

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py (hunk 2 above)
Comment (abc99lr) [reply tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py:91]: Thanks. Will add this as well as documentation

**Review by chang-l — COMMENTED** [review]
One additional test-maintenance note: the existing VisualGenParams tests use `_wan_mock(is_wan22=...)`, but this PR renamed the consumed flags to `is_wan22_14b` / `is_wan22_5b`. Since `_wan_mock` is a `MagicMock`, missing attributes are truthy, so the default/extra-param tests can accidentally exercise the 5B or 14B branches. Please update the helper/call sites to set the new flags explicitly, and add a small explicit 5B default case.

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py [hunk 3]
```diff
@@ -314,18 +325,24 @@ def _run_warmup(self, height: int, width: int, num_frames: int, steps: int) -> N
...
     def extra_param_specs(self):
-        return get_wan_extra_param_specs(self.is_wan22)
+        return get_wan_extra_param_specs(self.is_wan22_14b)
 
     def infer(self, req):
         """Run inference with request parameters."""
         extra = req.extra_params or {}
+        # Wan 2.2 TI2V-5B takes one conditioning image if provided
+        image = getattr(req, "image", None)
+        if isinstance(image, list):
```
Comment (chang-l) [inline tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py:350]: `VisualGenParams`/`DiffusionRequest` allow `image` to be a list, and the existing `WanImageToVideoPipeline` normalizes a list by taking the first image. This new TI2V path rejects any list, so a valid single-image request like `image=["/path/img.png"]` fails before reaching the 5B I2V implementation. Could we accept length-1 lists here and reject only multi-image batches?

**Review by chang-l — COMMENTED** [review]
One more small robustness point around the moved defaults.

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_utils.py (hunk 1 above)
Comment (chang-l) [reply tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_utils.py:20]: I prefer keep as-is to avoid unnecessary changes

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_i2v.py [hunk 4]
```diff
@@ -472,14 +454,7 @@ def forward(
...
 
-        # Set model-specific defaults based on Wan version
-        if num_inference_steps is None:
-            num_inference_steps = 40 if self.is_wan22 else 50
-
-        if guidance_scale is None:
-            guidance_scale = 4.0 if self.is_wan22 else 5.0
-
-        if self.is_wan22 and guidance_scale_2 is None:
+        if self.is_wan22_14b and guidance_scale_2 is None:
```
Comment (chang-l) [inline tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_i2v.py:464]: Instead of deleting the local fallback entirely here, can we make this more defensive, e.g., 
```
defaults = self.default_generation_params
if num_inference_steps is None:
    num_inference_steps = defaults["num_inference_steps"]
if guidance_scale is None:
    guidance_scale = defaults["guidance_scale"]
```

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py (hunk 2 above)
Comment (chang-l) [reply tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py:91]: Thank you, @abc99lr.

Maybe it is also worthwhile to document Wan2.2 TI2V-5B support in `docs/source/models/visual-generation.md` and `docs/source/models/supported-models.md`.

Comment (abc99lr) [reply tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py:91]: Added documentations and tests

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py (hunk 3 above)
Comment (abc99lr) [reply tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan.py:350]: Fixed

File: tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_i2v.py (hunk 4 above)
Comment (abc99lr) [reply tensorrt_llm/_torch/visual_gen/models/wan/pipeline_wan_i2v.py:464]: Fixed, thanks

File: examples/visual_gen/visual_gen_wan_i2v.py [hunk 5]
```diff
@@ -53,26 +53,41 @@ def parse_args():
...
+        "--width", type=int, default=None, help="Video width (default: auto-detect)"
+    )
+    parser.add_argument(
+        "--num_frames",
+        type=int,
+        default=None,
+        help="Number of frames to generate (default: auto-detect)",
+    )
+    parser.add_argument(
+        "--frame_rate",
```
Comment (chang-l) [inline examples/visual_gen/visual_gen_wan_i2v.py]: Do we need to add new arg `frame_rate` here? Can model defaults take care of the diff frame rate between different Wan variants?

File: examples/visual_gen/visual_gen_wan_i2v.py [hunk 6]
```diff
@@ -352,7 +378,9 @@ def main():
 
         logger.info(f"Generation completed in {time.time() - start_time:.2f}s")
 
-        MediaStorage.save_video(output.video, args.output_path, audio=output.audio, frame_rate=16.0)
+        MediaStorage.save_video(
+            output.video, args.output_path, audio=output.audio, frame_rate=frame_rate
```
Comment (chang-l) [inline examples/visual_gen/visual_gen_wan_i2v.py:410]: Same here — how about setting `frame_rate = defaults.frame_rate` so users don’t need to worry about specifying it in the example?

**Review by chang-l — APPROVED** [review]
Thanks for the contribution!

File: examples/visual_gen/visual_gen_wan_i2v.py (hunk 5 above)
Comment (abc99lr) [reply examples/visual_gen/visual_gen_wan_i2v.py]: Right now, it sets to model's default if user does not provide an override. I am fine with removing it too, if we'd like to reduce the number of options

Comment (abc99lr) [reply examples/visual_gen/visual_gen_wan_i2v.py]: Removed. We can add `frame_rate` option per user requests,

File: examples/visual_gen/visual_gen_wan_i2v.py (hunk 6 above)
Comment (abc99lr) [reply examples/visual_gen/visual_gen_wan_i2v.py:410]: Same as above
